### ISSUE TITLE
project panel: Allow showing both folder icons and open/closed arrows

### ISSUE THREAD

**Issue by y7ya-com** [zed-industries/zed#8661]
### Describe the feature

We need folder open/closed arrows. Icon-only approaches are very hard to navigate especially when someone has been used to folder arrows. Currently takes me around 20x longer to browse the explorer as I have to carefully analyse every single icon to know if it's a folder or not (combined with the fact that the icons are monochrome but that has been mentioned in other issues).

For someone who is used to folder arrows and/or colour coded icons, the editor is very hard to use & has a high barrier to entry without investing a lot of mental resources to get used to absence of these features.

### If applicable, add mockups / screenshots to help present your vision of the feature

<img width="194" alt="Screenshot 2024-03-01 at 11 14 36" src="https://github.com/zed-industries/zed/assets/78495264/67c7c35a-81ec-4c58-9206-f4933911cc92">

**Comment by Moshyfawn** [comment]
This can potentially be combined with #4915.

**Comment by JosephTLyons** [comment]
@yayashn, we do have this setting:

```json
  "project_panel": {
    "folder_icons": false
  },
```

Which removes the folder icon and uses an arrow:

<img width="387" alt="SCR-20240305-lnqp" src="https://github.com/zed-industries/zed/assets/19867440/18bb31fc-7736-424a-b859-afb879557afc">

But it does not show both the folder and the arrow; currently, only one can be configured to be shown.

Is this enough to satisfy what you're looking for?

**Comment by yayashn** [comment]
> @yayashn, we do have this setting:
> 
> ```json
>   "project_panel": {
>     "folder_icons": false
>   },
> ```
> 
> Which removes the folder icon and uses an arrow:
> 
> <img alt="SCR-20240305-lnqp" width="387" src="https://private-user-images.githubusercontent.com/19867440/310227102-18bb31fc-7736-424a-b859-afb879557afc.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDk2NjE0NTQsIm5iZiI6MTcwOTY2MTE1NCwicGF0aCI6Ii8xOTg2NzQ0MC8zMTAyMjcxMDItMThiYjMxZmMtNzczNi00MjRhLWI4NTktYWZiODc5NTU3YWZjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMDUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzA1VDE3NTIzNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTFkNjZmY2ViN2JjM2Q5OTRmN2FiM2FhNTc4ZTlmNTEwZDljOGZkMDBkMDNlMTlhYTg4MzA4ZGZlZTJlZTRlYzcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.38IMv-ia1IERrcTV3r9BcVHriTB5ThrrL9BYxmAjWb0">
> But it does not show both the folder and the arrow; currently, only one can be configured to be shown.
> 
> Is this enough to satisfy what you're looking for?

Well I wasn't expecting to have to disable folder icons to use the arrows, maybe I should update the issue to "allow folder arrows with file icons". But I assume the current available folder arrows are still technically folder icons made to look like arrows so might be worth it from a developer perspective to keep the issue as is for a real folder arrow implementation?

**Comment by Moshyfawn** [comment]
My 2 cents: add this information to #4915 and close this thread in favour of it.

**Comment by maxbrunsfeld** [comment]
I see that VS code has this by default: both arrows and icons.

I wouldn't want us to do this by default, since I think it's a bit visually noisy, but it seems reasonable to have a setting that lets you have both.

**Comment by fre-ben** [comment]
Push, really would appreciate an option to show arrows and icons at the same time.

**Comment by pslandyl** [comment]
It's so bad to only have folder icons without arrows.

**Comment by aolyang** [comment]
It's good to have an arrow.  It's hard to recognize an empty folder without arrow.  
I dont know the **folder is empty** or **not open** (my pain point).

**Comment by truehazker** [comment]
Yep, it's hard to get used to the absence of chevrons. I'm having problems distinguishing folders from files, especially migrating from VS Code to Zed

**Comment by hyuri** [comment]
+1
I think the value of showing the chevrons [in addition to folder icons] is twofold:
1. Aids in separating folders from files
2. Fulfills the expectation that there's more content under the element and therefore it can be expanded

Since the element is still a folder, it makes sense to show the folder icon too.

**Comment by Nemesis19** [comment]
recently landed in the zed world and I love it but this feature request is a super +1

**Comment by merchalex** [comment]
> coming from atom and vscode it's so hard to use folder icons without arrows in zed, folders just blend with file icons

exact same thing that led me to this thread. 

> recently landed in the zed world and I love it but this feature request is a super +1

 same, I'm one day 1 and this is killing me, I might go back to atom for a bit. one alternative that might work is putting the arrow icon in the folder icon png or something. obviously it wont have an open and closed state but maybe its better than nothing before they hopefully eventually add this.

**Comment by merchalex** [comment]
@Nemesis19 @truehazker @aolyang @fre-ben @vladshcherbin @pslandyl @y7ya-com 
for anyone else dealing with this issue, a good work around I found is the icon pack `colored zed icon themes dark`

no arrows showing open or closed, but all the folders are uniform and desaturated which is enough for me to be able to differentiate them from the other icons. 

<img width="649" height="743" alt="Image" src="https://github.com/user-attachments/assets/5fb66ee2-669d-4333-b3fb-81bcd2f51d6e" />

**Comment by aolyang** [comment]
Having used the Material Icon Theme for a few days, I have trained my brain to get used to it. 😆

**Comment by dongdong867** [comment]
@esthertrapadoux Can you assign this issue to me, thanks.
